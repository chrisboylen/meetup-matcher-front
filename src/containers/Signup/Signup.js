import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link, withRouter } from 'react-router-dom';
import { PropTypes } from 'prop-types';
import { auth, writeUserData } from '../../firebase/firebase';
import { createUserFirebase, updateUserFirebase } from '../../firebase/auth';
import { loginUser, userError } from '../../actions';

export class Signup extends Component {
  constructor() {
    super();
    this.state = {
      username: '',
      email: '',
      password: '',
      error: null
    }
  }

  handleChange = (e) => {
    const { name, value } = e.target;

    this.setState({ [name]: value })
  }

  handleSubmit = async (e) => {
    e.preventDefault();
    const { username, email, password } = this.state;
    const { login, history } = this.props;
    const newUser = {username, email};
    const userInfo = await createUserFirebase(email, password);
     await updateUserFirebase(username)

    if (!userInfo) {
      return userError('Email has already been taken.');
    }
    login(newUser);
    history.push('/user')
  }

  render() {
    const { username, email, password, error } = this.state;
    const isInvalid = username === '' || email === '' || password === '';

    return(
      <form 
        className="signup-form"
        onSubmit={this.handleSubmit}
      >
        <input 
          className="username-input-signup"
          required
          type="text"
          name="username"
          placeholder="Username"
          value={username}
          onChange={this.handleChange}
        />
        <input 
          className="email-input-signup"
          required
          type="email"
          name="email"
          placeholder="Email"
          value={email}
          onChange={this.handleChange}
        />
        <input 
          className="password-input-signup"
          required
          type="password"
          name="password"
          placeholder="Password"
          value={password}
          onChange={this.handleChange}
        />
        <button disabled={isInvalid}>Submit</button>
        {error && <h4>{error.message}</h4>}
      </form>
    )
  }
};

Signup.propTypes = {
  signup: PropTypes.func,
  userError: PropTypes.func
}

export const mapDispatchToProps = (dispatch) => ({
  login: (user) => dispatch(loginUser(user)),
  userError: (message) => dispatch(userError(message)) 
});

export default withRouter(connect(null, mapDispatchToProps)(Signup));