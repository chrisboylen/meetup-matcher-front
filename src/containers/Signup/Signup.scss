$mainDark: #280303;

.header-wrap {
  text-align: center;
}

.signup-form {
  background-color: #5c5252cc;
  border-radius: 10px;
  box-shadow: 0px 8px 19px 0px #0c090a;
  margin: 2rem;
  padding: 0;
  display: inline-flex;
  flex-direction: column;
  max-width: 40rem;
  min-width: 22rem;
    h1 {
      background-color: #fff5f5c7;
      border-radius: 10px 10px 0 0;
      color: $mainDark;
      margin: 0 0 1rem 0;
      padding: .5rem;
    }
    input, button {
      border-radius: 10px;
      margin: 1rem;
    }
}