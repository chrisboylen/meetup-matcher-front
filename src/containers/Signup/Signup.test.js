import React from 'react';
import { shallow } from 'enzyme';
import { Signup, mapStateToProps, mapDispatchToProps } from './Signup';

describe('Signup Component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Signup />);
  });

  it('should should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});