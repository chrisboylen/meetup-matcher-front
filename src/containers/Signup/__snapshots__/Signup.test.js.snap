// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup Component should should match snapshot 1`] = `
<div
  className="header-wrap"
>
  <withRouter(Connect(Header)) />
  <form
    className="signup-form"
    onSubmit={[Function]}
  >
    <h1>
      Signup
    </h1>
    <input
      className="username-input-signup"
      name="username"
      onChange={[Function]}
      placeholder="Username"
      required={true}
      type="text"
      value=""
    />
    <input
      className="email-input-signup"
      name="email"
      onChange={[Function]}
      placeholder="Email"
      required={true}
      type="email"
      value=""
    />
    <input
      className="password-input-signup"
      name="password"
      onChange={[Function]}
      placeholder="Password"
      required={true}
      type="password"
      value=""
    />
    <button
      disabled={true}
    >
      Signup
    </button>
  </form>
</div>
`;
